app/components/scheduled_post_indicator/scheduled_post_indicator.test.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Database} from '@nozbe/watermelondb';
+import {screen} from '@testing-library/react-native';
+import React from 'react';
+
+import NetworkManager from '@managers/network_manager';
+import {renderWithEverything} from '@test/intl-test-helper';
+import TestHelper from '@test/test_helper';
+
+import ScheduledPostIndicator from './';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+
+const SERVER_URL = 'https://appv1.mattermost.com';
+
+// this is needed when using the useServerUrl hook
+jest.mock('@context/server', () => ({
+    useServerUrl: jest.fn(() => SERVER_URL),
+}));
+
+describe('components/scheduled_post_indicator', () => {
+    let database: Database;
+    let operator: ServerDataOperator;
+
+    beforeEach(async () => {
+        const server = await TestHelper.setupServerDatabase(SERVER_URL);
+        database = server.database;
+        operator = server.operator;
+    });
+
+    afterEach(async () => {
+        await TestHelper.tearDown();
+        NetworkManager.invalidateClient(SERVER_URL);
+    });
+
+    it('should render single scheduled post indicator correctly', async () => {
+        const {getByText} = renderWithEverything(
+            <ScheduledPostIndicator
+                isThread={false}
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        await screen.findByTestId('scheduled_post_indicator_single_time');
+        expect(getByText(/Message scheduled for/)).toBeTruthy();
+        expect(getByText(/See all./)).toBeTruthy();
+    });
+
+    it('should render multiple scheduled posts indicator for channel', async () => {
+        const {getByText} = renderWithEverything(
+            <ScheduledPostIndicator
+                isThread={false}
+                scheduledPostCount={10}
+            />,
+            {database},
+        );
+
+        expect(getByText(/10 scheduled messages in channel./)).toBeTruthy();
+        expect(getByText(/See all./)).toBeTruthy();
+    });
+
+    it('should render multiple scheduled posts indicator for thread', async () => {
```

Comment:
duplicate with above?

Reply:
Nope, above one is for channels and this one is for threads. The text changes based on `isThread` prop, as being tested here.